### ISSUE TITLE
model comparison of better model evaluation

### ISSUE THREAD

**Issue by GreNait** [GreNait/technical-thesis#29]
Generate different modeld to evaluate their performence with different parameters:

- [x] https://www.tensorflow.org/tutorials/keras/overfit_and_underfit
- [ ] Train all models and make an comparison in a seperate document
- [ ] use the new tensorflow profiler https://www.youtube.com/watch?v=e4_4D7uNvf8 to get a better overview about the model effectiveness, integrate this finding into the document
- [ ] implementing an automised apporach for the trianing

**Comment by GreNait** [comment]
I generated different models (6 up to now) which are all similar but different in neurons and convolutional layers. The idea is to get a better overview of the capabilities of a model.

**Comment by GreNait** [comment]
![image](https://app.gitkraken.com/api/glo/boards/5e43bac3757b7a00112ba9d7/attachments/5e773f339585910011700b64)

**Comment by GreNait** [comment]
It appears, that something is completly off. I get from the beginning "nan" in the loss function.![image](https://app.gitkraken.com/api/glo/boards/5e43bac3757b7a00112ba9d7/attachments/5e7744b0e180cb0011db95f9)

**Comment by GreNait** [comment]
I made a heavy mistake, with the categorie in the default parameters. I have 3 figures and only provided 2 labels. Hterefore, the nan appeared. I changed that, and the model now starts training with a loss.![image](https://app.gitkraken.com/api/glo/boards/5e43bac3757b7a00112ba9d7/attachments/5e774c399585910011700c15)

**Comment by GreNait** [comment]
![image](https://app.gitkraken.com/api/glo/boards/5e43bac3757b7a00112ba9d7/attachments/5e7766b79585910011700e66)

**Comment by GreNait** [comment]
All three models (excluding non cnns) are able to detect somehow the figures. But they are not perfect. Antman is discovered as batman or vice versa. My guess is, that they all tend to overfitting therefore have problems with the recognition. I will try it again, this time with a dropout rate.

Normally, the first 4 should be a 0 (antman) -> cnn_small
![image](https://app.gitkraken.com/api/glo/boards/5e43bac3757b7a00112ba9d7/attachments/5e78d05be180cb0011dbc06b)

**Comment by GreNait** [comment]
Treined the big model with the Dropout, and the detection workded:

![image](https://app.gitkraken.com/api/glo/boards/5e43bac3757b7a00112ba9d7/attachments/5e78d913f602dd0018356da5)

**Comment by GreNait** [comment]
I found finally a way to get the steps per epoch right. The imageDtaGenerator has a way to provide that. Just use:
"imageDataGenerator2.samples. Thats it. TOTAL_TRAIN can than easily calculated by = self.train_data_generator.samples/self.train_data_generator.batch_size
